File: simple.cpp
#include <iostream>

int main() {

    int i = 0;
    for (int z = 1; ; ++z)
        for (int x = 1; x <= z; ++x)
            for (int y = x; y <= z; ++y)
                if (x*x + y*y == z*z) {
                    std::cout << "(" << x << "," << y << "," << z << ")" << std::endl;
                    if (++i == 1000)
                        goto done;
                }
    done:

    return 0;
}

File: simple.nim
include system/ansi_c

proc main() =
  var t = 0

  block done:
    var i = 0

    var z = 1
    while true:
      for x in 1..z:
        for y in x..z:
          if x*x + y*y == z*z:
            when defined(skip_printfs):
              t += x+y+z
            else:
              c_printf("(%i, %i, %i)\n", x, y, z)
            i.inc
            if i == 1000:
              break done
      z.inc

  when defined(skip_printfs):
    c_printf("%i\n", t)

main()
